### PR TITLE
New test for minpoly

### PR DESCRIPTION
**Pull request by david-lucas**
This PR proposes a test file for `minpoly` function.

### CODE REVIEW

File: tests/test-minpoly.C [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+	FFPACK::NonZeroRandomMatrix(F, 1, n, V, n, G); 
+	FFLAS::fassign(F, n, V, 1, Vcst, 1); //MatVecMinPoly modifies V, we store it in Vcst beforehand
+
+	FFPACK::MatVecMinPoly(F, minP, n, A, lda, V, ldv);
+	FFLAS::fflas_delete(V);
+
+	/*Check that minP is monic*/
+
+	size_t deg = minP.size() - 1;
+	if(!(minP[deg]==F.one))
```
Comment (ClementPernet) [inline tests/test-minpoly.C]: Do not use == but F.areEqual

File: tests/test-minpoly.C [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	size_t lda, ldv;
+	Element_ptr A, V, Vcst;
+
+	//Default
+	lda = n;
+	ldv = n; 
+
+	/*Create variables used for testing (matrices, vectors and polynomials) */
+
+    A = FFLAS::fflas_new(F, n, n);
```
Comment (ClementPernet) [inline tests/test-minpoly.C]: col dimension should be lda (in case we set lda to somehting > n later on

File: tests/test-minpoly.C [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    V = FFLAS::fflas_new(F, n+1, n);
```
Comment (ClementPernet) [inline tests/test-minpoly.C]: same thing here: ldv

File: tests/test-minpoly.C [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+	Vcst = FFLAS::fflas_new(F, 1, n);
+    Polynomial minP;
+
+
+    FFPACK::RandomMatrix (F, n, n, A, lda, G);
+
+	FFPACK::NonZeroRandomMatrix(F, 1, n, V, n, G);
```
Comment (ClementPernet) [inline tests/test-minpoly.C]: n->ldv for the 5th arg

File: tests/test-minpoly.C [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+	return ok;
+}
+
+
+int main(int argc, char** argv)
+{
+    /* Test parameters */
+	Givaro::Integer q = -1;
+	size_t b = 0;
+    size_t n = 128;
```
Comment (ClementPernet) [inline tests/test-minpoly.C]: Choose something not a power of 2. Larger would better (478)

**Comment by david-lucas** [comment]
I just pushed the requested changes. 
I chose a smaller number than the suggested 478 for your 5th comment as with 478, tests were way too slow.

David

**Pull request merged**